.github/workflows/test-semgrep-pro.yaml

```diff
@@ -0,0 +1,72 @@
+name: test-semgrep-pro
+
+on:
+  workflow_dispatch:
+    inputs:
+      artifact-name:
+        required: true
+        type: string
+        description: Name (key) to use when uploading the docker image tarball as a artifact
+      repository-name:
+        required: true
+        type: string
+        description: The repository/name of the docker image to push, e.g., returntocorp/semgrep
+  workflow_call:
+    inputs:
+      artifact-name:
+        required: true
+        type: string
+        description: Name (key) to use when uploading the docker image tarball as a artifact
+      repository-name:
+        required: true
+        type: string
+        description: The repository/name of the docker image to push, e.g., returntocorp/semgrep
+
+jobs:
+  test-semgrep-pro-with-pr:
+    name: Test Semgrep Pro Engine with PR version of Semgrep
+    if: ${{ github.event.pull_request.number != null }}
+    runs-on: ubuntu-22.04
+    env:
+      SEMGREP_APP_TOKEN: ${{ secrets.SEMGREP_APP_TOKEN }}
+    steps:
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+      - name: Download artifact
+        uses: actions/download-artifact@v3
+        with:
+          name: ${{ inputs.artifact-name }}
+          path: /tmp
+      - name: Load image
+        run: |
+          docker load --input /tmp/image.tar
+      - name: Run Semgrep Pro Engine!
+        run: |
+          docker run --rm -it -e SEMGREP_APP_TOKEN=${{ secrets.SEMGREP_APP_TOKEN }} --entrypoint bash "${{ inputs.repository_name }}:pr-${{ github.event.pull_request.number }}"
+
+          semgrep install-semgrep-pro
+          semgrep --config "p/deepsemgrep" . --interfile
+
+  test-semgrep-pro-with-develop:
+    name: Test Semgrep Pro Engine with develop version of Semgrep
+    if: ${{ github.event.pull_request.number == null }}
+    runs-on: ubuntu-22.04
+    env:
+      SEMGREP_APP_TOKEN: ${{ secrets.SEMGREP_APP_TOKEN }}
+    steps:
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+      - name: Download artifact
+        uses: actions/download-artifact@v3
+        with:
+          name: ${{ inputs.artifact-name }}
+          path: /tmp
+      - name: Load image
+        run: |
+          docker load --input /tmp/image.tar
+      - name: Run Semgrep Pro Engine!
+        run: |
+          docker run --rm -it -e SEMGREP_APP_TOKEN=${{ secrets.SEMGREP_APP_TOKEN }} --entrypoint bash "${{ inputs.repository_name }}:pr-${{ github.event.pull_request.number }}"
+
+          semgrep install-semgrep-pro
```

Comment:
This isn't going to do what you need it to, this will simply run the docker image (which'll then do whatever the default thing is, I think just a regular semgrep scan) and then will try to run semgrep outside the context of the running docker image. You might want to mount a script into the container, and run that script, or just do something of the form: `docker run ... <args> ... returntocorp/semgrep:pr-123 "semgrep install-semgrep-pro; semgrep --config p/deepsemgrep . --interfile"` 

Not sure that syntax works exactly, but happy to discuss if you want more details.